integration-tests/tests/src/tests/observable.ts

```diff
@@ -303,16 +372,15 @@ describe("Observable", () => {
     });
 
     it("calls listener", async function (this: RealmObjectContext<Person>) {
-      const completion = expectObjectNotifications(this.object, [
+      await expectObjectNotifications(this.object, [
         { deleted: false, changedProperties: [] },
+        () => {
+          this.realm.write(() => {
+            this.object.name = "Bob";
+          });
+        },
         { deleted: false, changedProperties: ["name"] },
       ]);
```

Comment:
Really neat idea with being able to see the actions and their effects sequentially like this 🙂